### DISCUSSION TITLE
authStore doesnt persist on page refresh after auth?

### DISCUSSION THREAD

**Discussion by jh97uk** [pocketbase/pocketbase#6103] [Q&A]
**OS:** Mac OS 15.1
**Browser:** Firefox 133.0
**Pocketbase version**: 0.22.1

**Issue:**
Whenever I login with `authWithPassword` (successfully), I can see that my `authStore` has all the associated data with the account I authed with... But upon refresh? The store is empty. I `console.log` the `authStore` whenever I type, and theres nothing there, so its not an event timing issue. But if I check the `LocalStorage` in my browser settings, I can see the token and the record?

**Expected behaviour:**
I should be able to authenticate with `authWithPassword`, and it retain the authentication upon refresh. It says on the [npm page for pocketbase](https://www.npmjs.com/package/pocketbase#localauthstore-default) that it defaults to `LocalStorage` and uses memory for fallback (thereby losing the auth on refresh?). So maybe the LocalStorage stuff is broken somehow? I've tried on both Firefox and Chrome 131.0.6778.140.

**Code:**
Its a React app. Heres my Pocketbase context:

```javascript
import React, { createContext, useContext, useRef, useState } from "react";

import PocketBase, { LocalAuthStore } from "pocketbase";

const PocketbaseContext = createContext();
const PocketbaseProvider = ({ children }) => {
  const pocketbase = new PocketBase(import.meta.env.VITE_PB_INSTANCE);
  pocketbase.autoCancellation(false);
  var login = (id, password) => {
    return pocketbase
      .collection("users")
      .authWithPassword(id, password, { autoRefreshThreshold: 30 * 60 });
  };
  return (
    <PocketbaseContext.Provider
      value={{
        login,
        authStore: pocketbase.authStore,
      }}
    >
      {children}
    </PocketbaseContext.Provider>
  );
};
const usePocketbase = () => useContext(PocketbaseContext);

export { PocketbaseProvider, usePocketbase };
```

Heres an example of where I use it:
```javascript
import React, {useState, useEffect} from 'react';
import { usePocketbase } from '../../PocketbaseContext';
function Login(props){
    const {login, authStore} = usePocketbase();
    const [formData, setFormData] = useState({
        id:'',
        password:''
    })
    const onChange = (target)=>{
        const {value, name} = target;
        let updatedFormData = {...formData};
        updatedFormData[name] = value;
        setFormData(updatedFormData);
    }
    const iLogin = ()=>{
        console.log(formData)
        login(formData['id'], formData['password']).then((result)=>{
            console.log(result);
        })
    }
    console.log(authStore);
    return <div className="centered-container">
        <div className="login-box">
            <form>
                <input name="id" placeholder="login ID" onChange={({target})=>onChange(target)}/>
                <input name="password" placeholder="password" type="password" onChange={({target})=>onChange(target)}/>
            </form>
            <button onClick={iLogin}>Login</button>
        </div>
    </div>
}

export default Login;
```

Any ideas would be appreciated. Many thanks!

**Comment by jh97uk** [comment]
A temporary fix for anyone who's experiencing the same as me would be using the `save` [function in `authStore`](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#common-auth-store-fields-and-methods) like so:

```javascript
 useEffect(() => {
    const authStoreJSON = localStorage.getItem("pocketbase_auth");
    if(authStoreJSON){
        const authStore = JSON.parse(authStoreJSON)
        pocketbase.authStore.save(authStore.token, authStore.record);
    }
  }, []);
```
This seems to work, but it's really not an optimal solution. I feel like I'm missing something very obvious?

If someone with more experience than me can confirm that I'm not just missing something, I can create an issue. If I get time tomorrow I'll recreate a vanilla JavaScript example to see if I can replicate the same behaviour.

Upon further investigation, without my fix, I can use the `authStore.onChange` callback, but it never gets called, despite having the LocalStorage entry to work with. No errors in the console either.

**Reply by sgraffite** [reply]
I ended up doing the same fix.  Not sure if this is on purpose and maybe they didn't want to assume you would want to resume using the existing token from local storage on page refresh?

**Reply by ganigeorgiev** [reply]
It is not a PocketBase issue and it could be easily tested with the following plain `pb_public/index.html`:

```html
<!DOCTYPE html />
<html>
<head>
    <meta charset="utf-8" />
    <title>Test</title>
</head>
<body>
    <script type="module">
        import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@master/dist/pocketbase.es.mjs"

        const pb = new PocketBase("http://127.0.0.1:8090");

        if (!pb.authStore.isValid) {
            console.log("new auth")
            await pb.collection("users").authWithPassword("test@example.com", "1234567890")
        } else {
            console.log("reusing old auth", pb.authStore.token)
        }
    </script>
</body>
</html>
```
Open the browser, navigate to `http://127.0.0.1:8090` and refresh twice the page (the first time you should see in the console _"new auth"_ and after that _"reusing old auth"_ and the token).

The problem could be either something with the above React setup (_I'm personally having difficulties understanding the purpose of the provided code in the first post_) or there is a browser extension that interfere with the load of the data from the `localStorage`.

**In all cases, I'd advice against copying random "workarounds" without having an understanding of the issue as it most likely hiding a bug somewhere.**

If anyone think that there is a PocketBase issue, feel to provide a **minimal** reproducible and I'll try to investigate it, but as of now it doesn't look like this.
